iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/mnode/container/CachedMNodeContainer.java

```diff
@@ -335,17 +326,17 @@ public synchronized void updateMNode(String name) {
     ICachedMNode node = removeFromMap(childCache, name);
     if (node != null) {
       if (updatedChildBuffer == null) {
-        updatedChildBuffer = new ConcurrentHashMap<>();
+        updatedChildBuffer = new MNodeUpdateChildBuffer();
       }
       updatedChildBuffer.put(name, node);
     }
   }
 
   @Override
   public synchronized void moveMNodeToCache(String name) {
-    ICachedMNode node = removeFromMap(newChildBuffer, name);
+    ICachedMNode node = getNewChildBuffer().removeFromFlushingBuffer(name);
     if (node == null) {
-      node = removeFromMap(updatedChildBuffer, name);
+      node = getUpdatedChildBuffer().removeFromFlushingBuffer(name);
```

Comment:
These two operation shall not be executed continuously. It should be split to two interfaces-- moveMNodeFromNewChildBufferToCache and modeMNodeFromUpdateChildBufferToCache. Each of the interface should be invoked based on the status in MemoryManager#VolatileSubtreeIterator.

Reply:
fixed